src/Types.hs

```diff
@@ -2,26 +2,46 @@
 {-# LANGUAGE OverloadedStrings #-}
 module Types where
 
-import Coinbase.Exchange.Types.Core
+import Coinbase.Exchange.Types.Core (ProductId (ProductId))
 import Coinbase.Exchange.MarketData (Candle)
-import Coinbase.Exchange.Types (ExchangeConf) 
+import Coinbase.Exchange.Types (ExchangeConf)
 
 newtype NumCandles = NumCandles Int
 newtype Minutes = Minutes Int
 
 -- Not sure what the difference between ratio and rational are and when I should use which ?
-newtype SMA = SMA Rational
+-- Another option is the "scientific" type from Data.Scientific ?
+type NumberRepresentation = Rational
+
+newtype SMA = SMA NumberRepresentation
+  deriving (Show, Eq, Num)
+newtype EMA = EMA NumberRepresentation
   deriving (Show, Eq, Num)
-newtype EMA = EMA Rational
+newtype Price = Price NumberRepresentation
   deriving (Show, Eq, Num)
 
 type CoinbaseCandle = Candle
 
-type Window = (EMA, EMA)
+data Action = Buy | Sell
+  deriving Show 
+data LookingTo = LookingTo Action 
+data Decision = Decision Action | Hold
+  deriving Show
+
+-- (Short, Long) ClosingPrice
+data Window = Window {
+  unEMAs :: (EMA, EMA),
+  unPrice :: Price }
+
+data World = World
+  { unMgr :: ExchangeConf
+  , unWindow :: Window }
 
-data World = World ExchangeConf Window
+instance Show World where
```

Comment:
I personally don't like `Show` instances that break the `read . show = show . read = id` law. In cases like these I find that either

* A `displayWorld :: World -> Text` function suffices or

* Something like `class Display a where display :: a -> Text` class works great

Large Haskell applications have a tendency to expose `show` strings to the end user, which I think is a poor use of the class because it's not designed to be the English representation of the datatype. Rather it's meant to be a string representation that is isomorphic to the original datatype.